### PR TITLE
mwocp68: add commands for firmware update

### PR DESCRIPTION
**Pull request by bcantrill**
Draft comments:

```
This adds some additional PMBus support for the MWOCP68-3600 series,
specifically around the commands necessary to support firmware update.
(This also performs some minor cleanup, bumping the Rust version
to 1.84 and fixing the resulting clippy warnings.)
```

### CODE REVIEW

**Comment by mkeeter** [comment]
The CI failures are because we've set an `MSRV` of 1.57 (Dec 2, 2021) in [`ci.yaml`](https://github.com/oxidecomputer/pmbus/blob/master/.github/workflows/ci.yaml), which doesn't support vars in format strings.  I'd be fine bumping that to something a little newer 😆

File: src/mwocp68.ron [hunk 1]
```diff
@@ -88,5 +97,87 @@
                 }),
             ),
         },
+
+        "MFR_PAGE_X": {
```
Comment (mkeeter) [inline src/mwocp68.ron:101]: Where did this come from?  I don't see it in the "PSU Firmware Update Process" doc or ACAN-114.

File: src/mwocp68.ron [hunk 2]
```diff
@@ -1,5 +1,13 @@
 (
-    all: [],
+    all: [
+        (0xe4, "MFR_PAGE_X", WriteByte, ReadByte),
+        (0xf0, "BOOT_LOADER_KEY", WriteBlock, Illegal),
+        (0xf1, "BOOT_LOADER_STATUS", Illegal, ReadBlock),
```
Comment (mkeeter) [inline src/mwocp68.ron:5]: No `BOOT_LOADER_WRITE` (which is also 0xf1)?

Comment (bcantrill) [reply src/mwocp68.ron:5]: We can't tolerate different commands with the same command code -- and nor should we. ;)  I'll add a comment explaining what's going on.

**Pull request merged**